great_expectations/rule_based_profiler/parameter_builder/parameter_builder.py

```diff
@@ -561,14 +564,25 @@ def _sanitize_metric_computation(
             for metric_value_idx in metric_value_indices:
                 metric_value: MetricValue = metric_values[metric_value_idx]
                 if enforce_numeric_metric:
-                    if not np.issubdtype(metric_value.dtype, np.number):
+                    if isinstance(metric_value, (str, np.str_)):
+                        if not is_parseable_date(value=metric_value):
+                            raise ge_exceptions.ProfilerExecutionError(
+                                message=f"""Applicability of {self.__class__.__name__} is restricted to numeric-valued \
+metrics (value {metric_value} of type "{str(type(metric_value))}" was computed).
+"""
+                            )
+                    elif not (
+                        isinstance(metric_value, datetime.datetime)
+                        or np.issubdtype(metric_value.dtype, np.number)
+                    ):
                         raise ge_exceptions.ProfilerExecutionError(
-                            message=f"""Applicability of {self.__class__.__name__} is restricted to numeric-valued metrics \
-(value of type "{str(metric_value.dtype)}" was computed).
+                            message=f"""Applicability of {self.__class__.__name__} is restricted to numeric-valued \
+metrics (value {metric_value} having numpy.dtype "{str(metric_value.dtype)}" and Python type \
+"{str(type(metric_value))}" was computed).
 """
                         )
 
-                    if np.isnan(metric_value):
+                    if pd.isnull(metric_value):
```

Comment:
curious: why the switch from `np` to `pd`?

Reply:
@Shinnnyshinshin Great question -- `np` is only for numbers, `pd` handles other types very well, too.